applications/swush-dex-aggregator.md

```diff
@@ -0,0 +1,281 @@
+- **Multi Asset Pair Swap**
+    - Support swaps between asset pairs not present in liquidity pools (e.g., MYTH/USDC when only DOT/USDC and DOT/MYTH are available).
+- **Optimized Routing**
+    - Automated route optimization to ensure that users are always getting the most cost-efficient swap by comparing fees, quote price and slippage across various DEXes(e.g. Polkadot Asset Hub DEX and Hydration DEX)
+- **Advanced Simulated Testing Environment:**
+    - Comprehensive testing of the entire DEX aggregator, including functionality and performance. This will involve simulating testing transaction reliability across multiple chains, and ensuring the stability of the platform under different scenarios (e.g., high transaction volume, fluctuating slippage).
+- **User Transaction History:**
+    - Implementation of user transaction history, enabling users to view their past swaps.
+
+Note: for parachain DEX integration we will do for Hydration only as of now. 
+
+ 
+
+| Number | Deliverable | Specification |
+| --- | --- | --- |
+| **0a.** | License | GPLv3  |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** |
+| **0c.** | Testing and Testing Guide | Tutorial that explains how to test all deliverables of this milestone. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an article for the work done. |
+| 1. | **Cross-Chain (XCM) APIs and DEX Swap Integration** | (as mentioned above) |
+| 2. | **Slippage Control and Protection** | (as mentioned above) |
+| 3. | **Real-time Fee Updates** | (as mentioned above) |
+| 4. | **Multi Asset Pair Swap** | (as mentioned above) |
+| 5. | **Optimized Routing** | (as mentioned above) |
+| 6. | **Advanced Simulated Testing Environment** | (as mentioned above) |
+| 7 | **User Transaction History** | (as mentioned above) |
+
+
+## Future Plans
+
+- **Long-term Maintenance and Development Funding:**
+    - With our estimated budget, we plan for the platform to be self-sustainable for maintenance and further development until it achieves product-market fit (PMF).
+    - Should we see significant user traction after Plaza’s full launch, especially with smart contract support, we will seek additional funding.
+- **Short-term Strategy:**
+    - Our initial focus is on a beautiful UX and strong branding to build user confidence in our platform. We recognize that trust-building will take time, as users may be cautious initially. To attract new users, we plan to launch NFTs and incentive programs that encourage engagement and usage.
+- **Long-term Vision:**
+    - Establish Swush as a trusted platform and cross-chain transaction hub for Polkadot.
+    - Achieve self-sustainability through revenue streams such as trading fees, liquidity provider (LP) fees, or yield farming.
```

Comment:
@takahser No such plans yet. From my understanding, DEX Aggregators typically have more limited features compared to standalone DEXs. They primarily focus on finding the best rates for asset swaps across various DEXs, as seen in examples like Jupiter on Solana.

Reply:
> From my understanding, DEX Aggregators typically have more limited features compared to standalone DEXs. They primarily focus on finding the best rates for asset swaps across various DEXs, as per my observations on DEX Aggregator like Jupiter on Solana.

That's correct. I thought it would still be a nice feature to have, just because I was assuming that no other dApp already provides this feature. However, I just realized that Kheoswap already [already allows](https://kheopswap.xyz/#/polkadot/pools/9) users to provide liquidity. Hence, I think it's fine to solely focus on the aggregation part for the scope of this grant.